### DISCUSSION TITLE
how do you get data from Clickhouse in Postgres

### DISCUSSION THREAD

**Discussion by hiepxanh** [ClickHouse/ClickHouse#41464] [Q&A] [answered]
as a developer, I have book table in Postgres with bookId, title, description. and have book_tracking have bookId, viewCount.

Now I want to get viewCount from clickhouse, then mix it with title, description then provide it in a materialized table in postgres. How can I do that?

example:
| POSTGRES TABLE |  CLICKHOUSE TABLE | EXPECT POSTGRES TABLE |
| --- | --- | --- |
| **books** | **books_tracking** | **mv_books** |
| bookId | bookId | bookId |
| title | view_count | title
| description | | description|
| | | view_count (from clickhouse) |

the first solution is: create Foreign Data Wrappers (FDW) (look like not stable, hard to setup)

second solution is: create a table, update it manually by cronjob. truncate it then fill it with data from clickhouse every 5 minute (I'm using this way)

is there any better solution?

**Comment by tbragin** [comment, marked as answer]
Fwiw, Supabase recently added their version Foreign Data Wrapper for ClickHouse and we jointly demoed it. You can see this blog for details and link to recording https://clickhouse.com/blog/adding-real-time-analytics-to-a-supabase-application

<img width="496" alt="Screenshot 2023-06-02 at 4 54 59 PM" src="https://github.com/ClickHouse/ClickHouse/assets/933397/f8949315-7fa7-4aac-a3ef-9ea08540ba3b">

**Reply by hiepxanh** [reply]
interesting, thank you for this information
